Add allow_external flag & Fix Tests

EdwardJXLi opened the pull request:
# Motivation

<!-- Why is this change necessary? -->

# Content

<!-- Please include a summary of the change -->

# Testing

<!-- How was the change tested? -->

# Please check the following before marking your PR as ready for review

- [ ] I have added tests for my changes
- [ ] I have updated the documentation or added new documentation as needed

kopekC committed:
misc: better-error (#805)

# Motivation

<!-- Why is this change necessary? -->

# Content

<!-- Please include a summary of the change -->

# Testing

<!-- How was the change tested? -->

# Please check the following before marking your PR as ready for review

- [ ] I have added tests for my changes
- [ ] I have updated the documentation or added new documentation as
needed

---------

Co-authored-by: kopekC <28070492+kopekC@users.noreply.github.com>

EdwardJXLi committed:
Add allow_external flag

EdwardJXLi committed:
Add docs for allow_external

EdwardJXLi committed:
Fix tests (cwd fix)

EdwardJXLi committed:
Automated pre-commit update

bagel897 commented on src/codegen/sdk/utils.py:
```diff
@@ -339,3 +339,29 @@ def is_minified_js(content):
+    """
+    old_cwd = os.getcwd()
+    try:
+        os.chdir(path)
```
Can we use this as a pytest fixture?

EdwardJXLi replied:
Hmm, looking into it!

bagel897 commented on docs/introduction/advanced-settings.mdx:
```diff
@@ -309,6 +309,19 @@ Controls import path overrides during import resolution.
+For this to properly work, you may also need to enable `allow_external`.
+</Note>
+
+## Flag: `allow_external`
```
Hm I don't think we have the performance in the current sdk to handle external resolution so this makes sense

Commit:
[CG-10935] fix: issues with assigment (#737)

# Motivation

<!-- Why is this change necessary? -->

# Content

<!-- Please include a summary of the change -->

# Testing

<!-- How was the change tested? -->

# Please check the following before marking your PR as ready for review

- [ ] I have added tests for my changes
- [ ] I have updated the documentation or added new documentation as
needed

---------

Co-authored-by: tomcodgen <191515280+tomcodgen@users.noreply.github.com>
Co-authored-by: tomcodegen <kucar.tomislav@gmail.com>

EdwardJXLi committed:
Upgrade allow_external warning to error

EdwardJXLi committed:
Remove use_cwd

EdwardJXLi committed:
Automated pre-commit update

EdwardJXLi committed:
Add JWT as explicit dependency (#810)

# Motivation

<!-- Why is this change necessary? -->

# Content

<!-- Please include a summary of the change -->

# Testing

<!-- How was the change tested? -->

# Please check the following before marking your PR as ready for review

- [ ] I have added tests for my changes
- [ ] I have updated the documentation or added new documentation as
needed

EdwardJXLi committed:
Finish merge

EdwardJXLi committed:
Merge branch 'develop' into eli-misc-assert-dir-fix-tests

EdwardJXLi commented:
I flew a little too close to the sun with git rebase
